sdk/core/Azure.Core/src/DynamicData/DynamicData.cs

```diff
@@ -20,6 +22,9 @@ namespace Azure.Core.Serialization
     /// This and related types are not intended to be mocked.
     /// </summary>
     [DebuggerDisplay("{DebuggerDisplay,nq}")]
+#if !NET5_0 // RequiresUnreferencedCode in net5.0 doesn't have AttributeTargets.Class as a target, but it was added in net6.0
+    [RequiresUnreferencedCode("This class utilizes reflection-based JSON serialization and deserialization which is not compatible with trimming.")]
```

Comment:
Can you use the `const` you defined elsewhere? Hopefully the compiler will pool the string anyway, but keeps it consistent so that string pooling is possible.